### PR TITLE
Fix #518 for 07-find

### LINKED ISSUE #518: Missing solutions to, context for, and files required by exercises
Many of the exercises don't have solutions. Some don't need solutions because learners can run the commands and see which is the correct answer. In some of the exercises, the commands can't be tested because the data files don't exist e.g. 07-find, "matching and subtracting" - there are no *ose.dat files

> The -v flag to grep inverts pattern matching, so that only lines which do not match the pattern are printed. Given that, which of the following commands will find all files in /data whose names end in ose.dat (e.g., sucrose.dat or maltose.dat), but do not contain the word temp?
> 
> find /data -name '*.dat' | grep ose | grep -v temp
> find /data -name ose.dat | grep -v temp
> grep -v "temp" $(find /data -name '*ose.dat')
> None of the above.

Sometimes it isn't clear which directory the question refers to.

I plan to submit a series of PRs to address these issues, using existing files as much as possible. Is this of interest to others?

### PR DESCRIPTION
**Pull request by gcapes**
Provide solutions and modify exercises to use the example data in data-shell.
Also add files and directories where using existing files wasn't possible.

### CODE REVIEW

**Comment by gcapes** [comment]
@shwina 
Just a bump, because there have been no comments on this yet.

File: _episodes/07-find.md [hunk 1]
```diff
@@ -514,20 +519,38 @@ about them."
 > wc -l $(find . -name '*.dat') | sort -n
 > ~~~
 > {: .bash}
+>
+> > ## Solution
+> > Find all files with a `.dat` extension in the current directory.
+> > Count the number of lines each of these files contains, then sort the output numerically.
+> {: .solution}
```
Comment (shwina) [inline _episodes/07-find.md]: How about we list (enumerate) them:

1. Find all files with a `.dat` extension in the current directory
2. Count the number of lines in each of the above files
3. Sort the output from step (2) numerically

File: _episodes/07-find.md [hunk 2]
```diff
@@ -579,13 +618,39 @@ about them."
...
+> > ```
+> > {: .source}
+> >
+> > Alternative, slightly inferior solution:
+> > ```
+> > for sis in Jo Meg Beth Amy
+> > do
+> > 	echo $sis:
+> >	grep -ocw $sis LittleWomen.txt
+> > done
```
Comment (shwina) [inline _episodes/07-find.md]: I'd say let's leave out the inferior solution altogether. Any advantage to keeping it?

Comment (gcapes) [inline _episodes/07-find.md]: Only that the question suggests two different ways to achieve the outcome. I don't mind removing the alternative solution, so long as we also remove the hint which suggests it.

Or I could add a short note explaining that there is often more than one way to complete a programming task.

Comment (shwina) [inline _episodes/07-find.md]: Thanks, @gcapes - I'd suggest keeping the hint and adding the note.

Comment (gcapes) [inline _episodes/07-find.md]: Done!

File: _episodes/07-find.md (hunk 1 above)
Comment (gcapes) [inline _episodes/07-find.md]: Done!

**Comment by shwina** [comment]
Thanks!

**Pull request merged**